python/plugins/processing/algs/qgis/RandomPointsAlongLines.py

```diff
@@ -116,14 +115,22 @@ def processAlgorithm(self, parameters, context, feedback):
 
         ids = source.allFeatureIds()
 
-        while nIterations < maxIterations and nPoints < pointCount:
+        while ids and nIterations < maxIterations and nPoints < pointCount:
             if feedback.isCanceled():
                 break
 
             # pick random feature
             fid = random.choice(ids)
-            f = next(source.getFeatures(request.setFilterFid(fid).setSubsetOfAttributes([])))
+            try:
+                f = next(source.getFeatures(request.setFilterFid(fid).setSubsetOfAttributes([])))
+            except:
```

Comment:
Can we tighten this exception ?

Reply:
I suppose this is the needed change `except StopIteration:`